UI Post-Refactor Revisions `[AARD-2002]`

itspvty opened the pull request:
## Task

<!--
Please include any relevant Jira ticket ID(s) at the end of the PR title, in the form AARD-xxxx, where "AARD" is Jira project.
Include the same Jira ticket ID(s) in this section.
-->

AARD-2002

This task involved collecting bugs relevant to the UI refactor from you guys and writing fixes for each one.

Task list:

- [x] Toasts have a "findDOMNode is deprecated" warning in devtools
- [x] The logo on the sidebar is draggable
- [x] Panels are too tall (even with no content)
- [x] The developer tool "No mira field loaded" red color is ugly, as well as some other red texts
- [x] You can open more than one unique panel
- [x] Panels on top of each other don't have proper separation (maybe add box shadow)
- [x] The "spawn" window and the "configure assets" windows shouldn't be able to be open at the same time
- [x] The dropdown shadows are too subtle/weird
- [x] Can't drag panels if pressing their title
- [x] The robot right-click context window shows the text "A Robot" in black at the top
- [x] Sliders do not display numerical values
- [x] There isn't enough spacing between many buttons (ex, match mode config menu)
- [x] The drag mode indicator at the bottom left has dark text on a dark bg
- [x] The timer on the scoreboard is dark text on dark bg
- [x] Where present, "delete" and "edit" buttons are placed weirdly (a line after their respective element often)
- [x] Add padding to the match mode config panel items
- [x] Block opening config/spawn when one of the two is already open
- [x] Fix slider overflow
- [x] Add info content to empty dropdowns
- [x] Pressing plus on the configure asset panel doesn't open the spawn assets panel
- [x] No feedback for why the config panel doesn't open when the spawn asset panel is open
- [x] Panels can be too tall
- [x] The configure panel for inputs skips to a specific scheme
- [x] The APS import folder doesn't have any text when empty
- [x] ~~Draggable region distinction~~
- [x] Fix analytics consent panel color
- [x] "Requires Browser Refresh" poorly styled; to tooltip
- [x] There are two scrollbars on tall panels
- [x] Blocked from closing the configure assets panel even with no configuration active
- [x] "Grab" cursor on draggable regions
- [x] Fix font sizes/logic throughout (titles/headers)
- [x] Fix spacing issues in more places
- [x] APS spinner alignment fix
- [x] Fix "submit" and "cancel" button overflow
- [x] Slider label/position uniformity
- [x] Fix input configure placements
- [x] Increase the maxHeight of all panels to 85vh
- [x] Make the panel title and button background the same color as the content
- [x] Keybind setters say "..." instead of "PRESS ANYTHING" when activated


## Symptom
The UI refactor was a large code edit and brought various visual bugs with it. The UI was, in most cases, fully usable, but the styling wasn't production-ready.

## Solution
Fixing individual issues over time by collecting purposeful data.

## Verification
Look good? Yay! Look bad? Aw man :(

---

Before merging, ensure the following criteria are met:

- [ ] All acceptance criteria outlined in the ticket are met.
- [ ] Necessary test cases have been added and updated.
- [ ] A feature toggle or safe disable path has been added (if applicable).
- [ ] User-facing polish:
  - Ask: *"Is this ready-looking?"*
- [x] Cross-linking between Jira and GitHub:
  - PR links to the relevant Jira issue.
  - Jira ticket has a comment referencing this PR.

itspvty committed:
UI post-refactor revisions pass 1

rutmanz commented:
- [ ] checkbox

PepperLola reviewed (changes requested):
Looks good overall but I noticed a couple things.

- I think the match mode config panel items could use some padding
- I'm not sure whether it makes more sense to close configure/spawn when the other is opened or to block the opening, because the user could be in the middle of configuring something and then opening spawn panel would cancel all their changes
- When dragging a slider in the settings modal all the way to the left or right the handle gets cut off, and on the right side it overflows and causes there to be a horizontal scroll bar
- I'm not sure how to address this, but I think the empty accordions in the spawn panel look a bit weird when expanded.

rutmanz commented:
> I'm not sure how to address this, but I think the empty accordions in the spawn panel look a bit weird when expanded.

Maybe just a message that is like "No Saved Assets" / "No Assets Found" rather than leaving it empty

rutmanz commented:
> I'm not sure whether it makes more sense to close configure/spawn when the other is opened or to block the opening, because the user could be in the middle of configuring something and then opening spawn panel would cancel all their changes

I think configure closing spawn is fine, but if configure is editing something (not just open) then spawn should show a warning

itspvty committed:
pass 2 fixes

itspvty committed:
pass 3 (pr recs)

itspvty committed:
pass 4

rutmanz commented:
1. I think that configure assets should also not be openable when the initial config panel (right after spawning) is open because it lets you configure the same robot in two different places which I don't think makes sense
2. The input screen doesn't look quite right still, and I also don't seem to be able to drag it anymore (use one of the controller input schemes)
<img width="5088" height="3726" alt="image" src="https://github.com/user-attachments/assets/6845f58a-bb2d-4193-b0e9-afb8d6380aa0" />


3. Reset orientation is kind of weird here, maybe just have one reset button? 
<img width="396" height="801" alt="image" src="https://github.com/user-attachments/assets/240b8ff1-157d-4ce5-9ed8-d5cd74c851ba" />


4. I don't think we always want to see values (particularly for scalars that don't actually have real-world meaning) on sliders. Also "Preferences" should be more of a header IMO
<img width="439" height="801" alt="image" src="https://github.com/user-attachments/assets/12bc08d2-6ef8-4315-9121-49f328d8b80d" />

rutmanz reviewed (changes requested):
see above ^

itspvty committed:
pass 5 (pr recs)

itspvty committed:
chore: rm unused

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

AlexD717 commented on fission/src/ui/ThemeProvider.tsx:
```diff
@@ -45,6 +45,33 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
+                            boxShadow: "0 4px 12px rgba(0,0,0,0.28), 0 2px 6px rgba(0,0,0,0.18)",
+                        },
+                    },
+                },
```
What do you think about adding something like this? This blurs the background if you have a modal open which I think is better.
```suggestion
                },
                MuiBackdrop: {
                    styleOverrides: {
                        root: {
                            backgroundColor: "rgba(0, 0, 0, 0.5)",
                            backdropFilter: "blur(2px)",
                        },
                    },
                },
```

rutmanz reviewed (changes requested):
- [x] Pressing the plus button on the configure asset panel doesn't open the spawn assets panel anymore
- [x] There is no feedback for why the config panel doesn't open when the spawn asset panel is open. I think it makes more sense to make the newly-opened one replace the other, unless you're editing a configuration, at which point it should show a warning toast
- [x] Panels can still be too tall

<img width="439" height="1073" alt="Image" src="https://github.com/user-attachments/assets/5dbba4e4-0422-4241-bf92-28f95d28979e" />

- [x] The configure panel for inputs seems to skip to a specific scheme instead of showing the list (and there are no breadcrums)
- [x] The APS import folder doesn't have any text when empty
- [x] If only parts of a panel are going to be draggable, there should be a visible distinction (maybe a handle like [this](https://react-icons.github.io/react-icons/search/#q=TbGridDots))

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

itspvty committed:
pass 6 (pr reqs)

PepperLola reviewed (changes requested):
The analytics consent panel text is a weird color and the buttons (at least the close button) need fixing.
<img width="405" height="207" alt="Image" src="https://github.com/user-attachments/assets/b825c90a-73b0-47b4-9e43-4f7b5ff77511" />

itspvty committed:
fix consent screen styling

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

rutmanz commented on fission/src/ui/components/Panel.tsx:
```diff
@@ -57,21 +56,54 @@ export const Panel = <T, P>({ children, panel, parent }: PanelElementProps<T, P>
+                {props.title && (
+                    <CardHeader
+                        title={props.title}
+                        className="panel-drag-handle select-none"
```
```suggestion
                        className="panel-drag-handle select-none hover:cursor-grab active:cursor-grabbing"
```

rutmanz commented on fission/src/ui/components/Panel.tsx:
```diff
@@ -57,21 +56,54 @@ export const Panel = <T, P>({ children, panel, parent }: PanelElementProps<T, P>
+                                        "linear-gradient(90deg, rgba(255,255,255,0.2) 25%, rgba(0,0,0,0.2) 25%, rgba(0,0,0,0.2) 50%, rgba(255,255,255,0.2) 50%, rgba(255,255,255,0.2) 75%, rgba(0,0,0,0.2) 75%)",
+                                    backgroundSize: "8px 100%",
+                                }}
+                            />
```
ok so having seen how this looks, I think it's clearer to just use different cursor formats to indicate where it's draggable (suggestion above)

rutmanz reviewed (changes requested):
- [x] The configure panel for inputs seems to skip to a specific scheme instead of showing the list
  - Spawn Dozer
  - Go to configure assets > inputs
  - See that it shows the control scheme for the robot you just spawned rather than the list
- [x] The graphics settings "Requires Browser Refresh" thing is not styled as a header (which makes it look like another setting) and makes more sense just as a tooltip on anti-aliasing IMO
- [x] There are [two scroll bars](https://github.com/user-attachments/assets/3ecb58bf-d8b2-40cb-a155-cfbbbe84679a) on tall panels (configure inputs is where I found it)
- [ ] Blocks me from closing the configure assets panel even when there is no configuration active [img](https://github.com/user-attachments/assets/bce85bcf-9cd0-42ad-b7f6-6af24be7d94d)

itspvty committed:
Update fission/src/ui/components/Panel.tsx

Co-authored-by: Zach Rutman <zarutman+git@gmail.com>

itspvty committed:
pass 7 (pr reqs)

itspvty committed:
Merge remote-tracking branch 'refs/remotes/upstream/aries/2002/fix-ui-appearance' into aries/2002/fix-ui-appearance

itspvty committed:
chore: format

itspvty committed:
remove action visual following grabber cursor implementation

ryanzhangofficial reviewed (dismissed):
Two issues: 

1) Font sizes/logic throughout. For example, in the attached image, the title "Assembly Setup" is so small compared to the two fields below it "Alliance" and "Station", which looks odd visually.

2) I don't think the button spacing has been fully addressed. It has been for the play buttons in match mode config. But not Create Match Mode Config and Upload File. I'm not sure if debug tools will still be in prod, but those buttons also seem cramped together.

I think Issue 2 can be addressed in a later ticket, but Issue 1 are changes in this PR that have made the UI worse.

<img width="626" height="1426" alt="Image" src="https://github.com/user-attachments/assets/a1102041-3214-43f1-abf9-3ae0ed3733b0" />

PepperLola reviewed (changes requested):
- I would like the APS spinner to be either the same height as the text (ideal) or centered vertically.

<img height="438" alt="Image" src="https://github.com/user-attachments/assets/fcd87e69-f796-466e-80a4-805d3563512e" />

- The default height of the initial config panel makes the accept/cancel buttons overflow. Maybe a worthwhile change would be making the modal/panel content scrollable but keeping the buttons visible (and the title as well)

<img height="438" alt="Image" src="https://github.com/user-attachments/assets/b182762f-d92e-4250-8046-ba035bd9cbf3" />

- If it's not too difficult it would be nice to have the slider labels match the other ones (in size + alignment/indentation); right now they look a bit out of place

<img height="438" alt="Image" src="https://github.com/user-attachments/assets/ba25e75c-fa5d-4eea-a0a4-a94705f0c50e" />

- This toggle group should be centered horizontally like in the spawn assets panel, and the buttons shouldn't touch the bottom of the panel (latter would be fixed by always showing accept/cancel)

<img height="438" alt="Image" src="https://github.com/user-attachments/assets/157e3f68-badc-41ea-a653-666f6473c091" />

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

rutmanz reviewed (approved):
Approving because you've addressed my previous review and everybody else has caught my remaining feedback

itspvty committed:
pass 8 (reqs)

itspvty committed:
fix typo

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

itspvty committed:
chore: format

AlexD717 commented:
How hard would it be for you to go through and increase the maxHeight of all panels to something closer to 85vh? I think that would be nicer.

PepperLola reviewed (changes requested):
- Can the panel title and button background be the same color as the content? I prefer the lighter background color between the two of them
- Can we make the input buttons say "..." when you click them rather than "PRESS ANYTHING"?

itspvty committed:
pass 9 (reqs)

itspvty committed:
chore: formatting

itspvty committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

PepperLola reviewed (changes requested):
Can the sliders here be stacked vertically?

<img width="982" height="642" alt="Image" src="https://github.com/user-attachments/assets/fa6069c8-ddeb-4553-8b72-3414f9d53c2b" />

The title and buttons have the old background in modals

<img width="982" height="642" alt="Image" src="https://github.com/user-attachments/assets/d900be71-a95f-4ab4-8186-8e3bbd34faa3" />

Can we switch the various configure panels to use tabs instead of the toggle group?

<img width="1894" height="642" alt="Image" src="https://github.com/user-attachments/assets/42cbca15-02a7-4318-bbd8-59a353543dec" />

There are two intake duration sliders

<img width="428" height="642" alt="Image" src="https://github.com/user-attachments/assets/496bbeb2-86f6-4098-9c21-08e422dc44a0" />

PepperLola reviewed (changes requested):
The requested changes were not addressed.

PepperLola committed:
fix: ui appearance changes
- updated main hud with new buttons, made button group
- made modal colors consistent with panel
- switched to using tabs instead of toggle button groups
- fixed aps reload button disappearing on import panel reopen

PepperLola committed:
Merge branch 'dev' into aries/2002/fix-ui-appearance

PepperLola committed:
fix: format and lint

PepperLola reviewed (approved):
Looks good
